lib/libzfs/libzfs_sendrecv.c

```diff
@@ -959,6 +963,22 @@ send_progress_thread(void *arg)
 		(void) time(&t);
 		localtime_r(&t, &tm);
 
+		if (pa->pa_astitle) {
+			char buf_bytes[16];
+			char buf_size[16];
+			int pct;
+			zfs_nicenum(bytes, buf_bytes, sizeof (buf_bytes));
+			zfs_nicenum(pa->pa_size, buf_size, sizeof (buf_size));
+			pct = (total > 0) ? bytes / total : 100;
+#ifdef __linux__
+			zfs_setproctitle("sending %s (%d%%: %s/%s)",
+			    zhp->zfs_name, MIN(pct, 100), buf_bytes, buf_size);
+#else
+			setproctitle("sending %s (%d%%: %s/%s)",
+			    zhp->zfs_name, MIN(pct, 100), buf_bytes, buf_size);
```

Comment:
Thanks for the quick update!  Let's take this one small step further a provide `zfs_setproctitle()` wrapper which calls `setproctitle()` for FreeBSD, and then a `zfs_setproctitle_init()` wrapper which is a no-op.  That will get these remaining #ifdef's out of the common code and set things up so it's easier to do the right thing on other platforms.

Reply:
Thanks, @behlendorf. I tried to do that in the first place (by wrapping FreeBSD functions in a new file ```lib/libzutil/os/freebsd/zutil_setproctitle.c``` just like did for Linux) but couldn't find a clean way. ```setproctitle(fmt, ...)``` contains variable arguments and a 'C' function does not allow passing through variable arguments. One way to do that is to wrap in a MACRO function instead, like ```#define zfs_setproctitle(fmt, ...) setproctitle(fmt, ##__VA_ARGS__)```, but we currently don't have platform-specific header files for libzutil. Let me know if there's a way to do this cleanly.

Reply:
I'd be fine with
```
#ifdef __FreeBSD__
#define zutil_setproctitle setproctitle
#else
/* prototype for linux */
#endif
```
in libzutil.h, or we could hide it in new OS-specific headers, but where to put them...

Reply:
Thanks, @freqlabs. I added it in ```libzutil.h``` to keep it simple.